src/main/java/crackerpackage/Cracker.java

```diff
@@ -0,0 +1,128 @@
+package crackerpackage;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import crackerpackage.tasks.Task;
+import exceptions.EmptyDescriptionException;
+import exceptions.IllegalFormatException;
+import exceptions.UnknownCommandException;
+import uicomponents.Parser;
+import uicomponents.Reply;
+
+/**
+ * A chatbot that manages tasks.
+ *
+ * @author Anton Tan Hong Zhi
+ */
+public class Cracker {
+
+    private TodoList list = null;
+    private Reply reply = new Reply();
+    private Storage storage;
+
+    /**
+     * The types of operations supported by the chatbot.
+     */
+    public enum Type {
+        MARK,
+        UNMARK,
+        TASK,
+        DELETE,
+        UNKNOWN,
+        LIST,
+        QUIT,
+        FIND
+    }
+
+    /**
+     * Starts up the bot.
+     */
+    public void startService() {
```

Comment:
I like how you used startService as the name. It relates well to the concept of a chatbot.